packages/ui/components/form/selectimproved/components/Options.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useCallback, useContext, useEffect, useMemo, useRef, useState } from "react";
+
+import { useKeyPress } from "@calcom/lib/hooks/useKeyPress";
+
+import { Label } from "../../inputs/Label";
+import Item from "./Item";
+import { SelectContext } from "./SelectProvider";
+import { Option } from "./type";
+
+interface OptionsProps<T extends Option> {
+  list: T[];
+  inputValue: string;
+  isMultiple: boolean;
+  selected: T | T[] | null;
+  searchBoxRef: React.RefObject<HTMLInputElement>;
+}
+
+type FlattenedOption = Option & { current: number; groupedIndex?: number };
+
+const flatternOptions = (options: Option[], groupCount?: number): FlattenedOption[] => {
+  return options.reduce((acc, option, current) => {
+    if (option.options) {
+      return [...acc, ...flatternOptions(option.options, current + (groupCount || 0))];
+    }
+    return [...acc, { ...option, current, groupedIndex: groupCount }];
+  }, [] as FlattenedOption[]);
+};
+
+function FilteredItem<T extends Option>({
+  index,
+  keyboardFocus,
+  item,
+  inputValue,
+  list,
+}: {
+  index: number;
+  keyboardFocus: number;
+  item: FlattenedOption;
+  inputValue: string;
+  list: T[];
+}) {
+  const focused = index === keyboardFocus;
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (ref.current && focused) {
+      ref.current.scrollIntoView({ behavior: "smooth" });
+    }
+  }, [ref, focused]);
```

Comment:
Since we're in a overflow-y-auto scrollIntoView scrolls the current item into view inside of that container :) 

Making it so you always can see the item you are hovering over

Reply:
This works great now 🫶